src/utils/streams/promise_from_streams.js

```diff
@@ -0,0 +1,67 @@
+/**
+ *  Take an array of streams, pipe the output
+ *  from each one into the next, listening for
+ *  errors from any of the streams, and then resolve
+ *  the promise once the final stream has finished
+ *  writing/reading.
+ *
+ *  If the last stream is readable, it's final value
+ *  will be provided as the promise value.
```

Comment:
It's used constantly in the tests with the concat stream, which always produces a single value. It's also the behavior of the `Observable#toPromise()` function in RxJS.